starter-code/scripts/models/article.js

```diff
@@ -25,27 +27,45 @@ Article.prototype.toHtml = function(scriptTemplateId) {
  call these "class-level" functions, that are relevant to the entire "class"
  of objects that are Articles, rather than just one instance. */
 
-/* TODO: Refactor this code into a function for greater control.
+/* TODO DONE: Refactor this code into a function for greater control.
     It will take in our data, and process it via the Article constructor: */
 
-ourLocalData.sort(function(a,b) {
-  return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
-});
-
-ourLocalData.forEach(function(ele) {
-  articles.push(new Article(ele));
-});
-
+Article.loadAll = function(dataWePassIn) {
+  dataWePassIn.sort(function(a,b) {
+    return (new Date(b.publishedOn)) - (new Date(a.publishedOn));
+  }).forEach(function(ele) {
+    Article.all.push(new Article(ele));
+  });
+};
 
 /* This function below will retrieve the data from either a local or remote
  source, process it, then hand off control to the View: */
 
-
-
-
-
-
-
+Article.fetchAll = function() {
+  if (localStorage.hackerIpsum) {
+    // when our data is already in local Storage we will first process (sort & instantiate) it, then render the index page
+    var localHacker = JSON.parse(localStorage.hackerIpsum);
+    Article.loadAll(localHacker);
+    articleView.renderIndexPage();
+
+     // TODO: DONE Invoke with our localStorage! Should we PARSE or stringify it?)
+    // TODO: DONE Now let's render the index page
+  } else {
+    /* TODO DONE: Otherwise, wothout our localStorage data, we need to:
+      1. Retrive our JSON file asynchronously
+      (which jQuery method is best for this?).
+      Within this method, we should:
+      1. Load our Json data
+      2. Store that data in localStorage so we can skip the erver call next time.
+      3. And then render the index page */
+
+    $.getJSON('data/hackerIpsum.json', function(data){
+      localStorage.hackerIpsum = JSON.stringify(data);
+      Article.loadAll(data);
+      articleView.renderIndexPage();
+    });
+  }
```

Comment:
Since you're doing `Article.loadAll(data)` and `articleView.renderIndexPage()` whether or not there's data in local storage, you should refactor so that you only do it once.